src/one/nio/http/HttpClient.java

```diff
@@ -234,6 +239,20 @@ Response readResponse(int method) throws IOException, HttpException {
             return response;
         }
 
+        String readResponseHeader() throws HttpException {
+            byte[] buf = this.buf;
+            int pos = 11;
+
+            do {
+                if (pos >= length) {
+                    throw new HttpException("Response header absent");
+                }
+            } while (buf[pos++] != '\n');
+
+            this.pos = pos;
+            return Utf8.read(buf, 9, pos - 2);
+        }
```

Comment:
I'm not sure why this function has been added, but it definitely looks incorrect. There is no guarantee that the first line of the response will be received entirely in the first packet.

Reply:
But how I can see in code: first line always get as is and then it's cut on `.substring(9)`
Why we need extra String? If you always cut it, then, why you need to create it with the first 9 bytes?

Reply:
This is just for simplicity and for code reuse. If we see this part a performance bottleneck, we'll optimize it.